src/lib/sections/ContentSection/ContentSection.tsx

```diff
@@ -57,8 +62,18 @@ const Title = ({ children, className, as, ...props }: ContentSectionProps) => {
   );
 };
 
-const Header = ({ children, className }: CommonContentSectionProps) => (
-  <header className={classNames("content-section__header", className)}>
+const Header = ({
+  children,
+  className,
+  isPinned,
+}: ContentSectionHeaderProps) => (
+  <header
+    className={classNames(
+      "content-section__header",
+      { "is-pinned": isPinned },
```

Comment:
_TLDR;  Let's put this on hold while we investigate how to resolve this._

Sorry, I know this is what I've suggested, but I've given this more thought and think we should simply make the header pinned by default (just as the footer is). 

Introducing a boolean prop could make this confusing and will be inconsistent with the footer defaults. I went through all instances in MAAS UI and do not see any disadvantages of using the sticky header in all places where this component is used.

One exception is mobile - sticky header doesn't work well when there's a lot of content on mobile as seen below.

![Google Chrome screenshot 001577@2x](https://github.com/canonical/maas-react-components/assets/7452681/00016ce1-cce6-4646-ba50-a2752e18a2ba)

I'd risk making the header sticky on large breakpoint only, but this could further complicate things.

In case we go ahead with it we should make sure we update the docs as well. Meaning - change from `ContentSection.Footer is made sticky by default.` to `Both ContentSection.Header and ContentSection.Footer are made sticky by default.`